app/scripts/main.js

```diff
@@ -392,6 +401,78 @@ var iViz = (function(_, $, cbio, QueryByGeneUtil, QueryByGeneTextArea) {
       }
       return def.promise();
     },
+    getScatterData: function(groupId) {
+      var def = new $.Deferred();
+      var self = this;
+      var data = self.getGroupNdx(groupId);
+
+      $.when(window.iviz.datamanager.getStudyToSampleToPatientdMap()).then(function(_studyToSampleToPatientMap) {
+        $.when(window.iviz.datamanager.getCaseLists()).then(function(_caseLists) {
+          var _sequencedCaseUIDs = [];
+          var _allCaseUIDs = [];
+          var _sequencedCaseUIdsMap = {};
+          
+          $.each(_caseLists, function(studyId, caseList) {
+            if (caseList.sequencedSampleIds.length > 0) {
+              $.each(caseList.sequencedSampleIds, function(index, sampleId) {
+                _sequencedCaseUIDs.push(_studyToSampleToPatientMap[studyId].sample_to_uid[sampleId]);
+              });
+            }
+            if (caseList.allSampleIds.length > 0) {
+              $.each(caseList.allSampleIds, function(index, sampleId) {
+                _allCaseUIDs.push(_studyToSampleToPatientMap[studyId].sample_to_uid[sampleId]);
+              });
+            }
+          });
+          _sequencedCaseUIDs = _sequencedCaseUIDs.length > 0 ? _sequencedCaseUIDs : _allCaseUIDs;
+          _.each(_sequencedCaseUIDs, function(_sampleUid) {
+            _sequencedCaseUIdsMap[_sampleUid] = _sampleUid;
+          });
+
+          $.when(window.iviz.datamanager.getCnaFractionData(),
+            window.iviz.datamanager.getMutationCount())
+            .then(function(_cnaFractionData, _mutationCountData) {
+              var _hasCNAFractionData = _.keys(_cnaFractionData).length > 0;
+              var _hasMutationCountData = _.keys(_mutationCountData).length > 0;
+
+              _.each(data, function(_sampleDatum) {
+                // mutation count
+                if (_hasMutationCountData) {
+                  if (_mutationCountData[_sampleDatum.study_id] === undefined ||
+                    _mutationCountData[_sampleDatum.study_id][_sampleDatum.sample_id] === undefined ||
+                    _mutationCountData[_sampleDatum.study_id][_sampleDatum.sample_id] === null) {
+                    if (_sequencedCaseUIdsMap[_sampleDatum.sample_uid] === undefined) {
+                      _sampleDatum.mutation_count = 'NA';
+                    } else {
+                      _sampleDatum.mutation_count = 0;
+                    }
+                  } else {
+                    _sampleDatum.mutation_count = _mutationCountData[_sampleDatum.study_id][_sampleDatum.sample_id];
+                  }
+                  mutationCountMap_[_sampleDatum.sample_id] = _sampleDatum.mutation_count;
```

Comment:
If you need to cachemutationCount, then you should use sample_uid instead of sample_id